### PR TITLE
UI Settings In a Seperate File

### PR DESCRIPTION
**Pull request by avigupta33**
Let's try this again!

Built out UI Settings into a separate file.

### CODE REVIEW

File: ovve_ui.py [hunk 1]
```diff
@@ -10,48 +10,51 @@
...
+        numberFont = self.button_settings.numberFont
+        unitFont = self.button_settings.unitFont
 
-        painter.setBrush(QBrush(QColor("#d2fcdc")))
+        painter.setBrush(self.button_settings.getFillBrush())
         painter.drawRect(0, 0, *self.size)
-        painter.setPen(QPen(QColor("#3ed5f0")))
+        painter.setPen(self.button_settings.getLabelPen())
         painter.setFont(labelFont)
         painter.drawText(int(self.size[0] / 2 - 50), int(self.size[1] / 5),
```
Comment (BrianWoodruff) [inline ovve_ui.py]: I think you could create a function in button settings to calcuate the x,y position of the text.

File: ovve_ui.py [hunk 2]
```diff
@@ -64,19 +67,20 @@ def updateValue(self, value):
 
 
 class SimpleDisplayButton(QAbstractButton):
-    def __init__(self, value, parent=None, size=(200, 50)):
+    def __init__(self, value, button_settings, parent=None, size=(200, 50)):
```
Comment (BrianWoodruff) [inline ovve_ui.py]: Can button_settings also contain the size?

File: ovve_ui.py [hunk 3]
```diff
@@ -90,38 +94,42 @@ def updateValue(self, value):
...
 
 class DisplayRect(QWidget):
-    def __init__(self, label, value, unit, parent=None, size=(200, 100)):
+    def __init__(self,
+                 label,
+                 value,
+                 unit,
+                 rect_settings,
+                 parent=None,
+                 size=(200, 100)):
```
Comment (BrianWoodruff) [inline ovve_ui.py]: Can rect_settings contain the size?

File: ovve_ui.py [hunk 4]
```diff
@@ -90,38 +94,42 @@ def updateValue(self, value):
...
-        painter.setPen(QPen(QColor("#3ed5f0")))
+        painter.setPen(self.rect_settings.getLabelPen())
         painter.setFont(labelFont)
         painter.drawText(int(self.size[0] / 2 - 50), int(self.size[1] / 5),
                          self.label)
-        painter.setPen(QPen(Qt.black))
+        painter.setPen(self.rect_settings.getNumberPen())
         painter.setFont(numberFont)
-        painter.drawText(int(self.size[0] / 2 - 50), int(3 * self.size[1] / 5),
+        painter.drawText(int(self.size[0] / 2 - 50), int(self.size[1] * 3 / 5),
```
Comment (BrianWoodruff) [inline ovve_ui.py]: Get x,y from rect_settings

File: ovve_ui.py [hunk 5]
```diff
@@ -189,50 +245,50 @@ def initializeWidget1(self):  # home screen
...
+        self.ppeak_display_main = self.makeDisplayRect("Ppeak",
+                                                       self.params.ppeak,
+                                                       "cmH2O",
+                                                       size=(150, 80))
+        self.pplat_display_main = self.makeDisplayRect("Pplat",
+                                                       self.params.pplat,
+                                                       "cmH2O",
+                                                       size=(150, 80))
+
+        axisStyle = {'color': 'black', 'font-size': '20pt'}
```
Comment (BrianWoodruff) [inline ovve_ui.py]: Can all the sizes be determined by makeDisplayRect?

File: ovve_ui.py [hunk 6]
```diff
@@ -405,24 +462,22 @@ def initializeWidget5(self):  # ie ratio
         h_box_5mid = QHBoxLayout()
         h_box_5bottom = QHBoxLayout()
 
-        self.ie_page_rect = DisplayRect("I/E Ratio",
-                                        self.settings.get_ie_display(),
-                                        "",
-                                        size=(500, 200))
+        self.ie_page_rect = self.makeDisplayRect(
+            "I/E Ratio", self.settings.get_ie_display(), "", size=(500, 200))
```
Comment (BrianWoodruff) [inline ovve_ui.py:331]: Size can be a property of ie_display

**Review by BrianWoodruff — CHANGES_REQUESTED** [review]

File: ovve_ui.py (hunk 3 above)
Comment (avigupta33) [reply ovve_ui.py]: My intention was that the rect_settings were unified across the entire UI to allow for styling all boxes at once. That was why I omitted size.

File: ovve_ui.py (hunk 2 above)
Comment (avigupta33) [reply ovve_ui.py]: My intention was that the button_settings were unified across the entire UI to allow for styling all boxes at once. That was why I omitted size.

**Pull request merged**